src/time/src/mcp_server_time/server.py

```diff
@@ -0,0 +1,145 @@
+from mcp.server import Server
+from mcp.server.stdio import stdio_server
+from mcp.types import Tool, TextContent, ImageContent, EmbeddedResource
+
+
+class TimeServer:
+    def __init__(self, local_tz_override: Optional[str] = None):
+        self.local_tz = pytz.timezone(local_tz_override) if local_tz_override else get_localzone()
+
+    def get_current_time(self, timezone_name: str | None = None) -> Dict[str, Any]:
+        """Get current time in specified timezone or local timezone if none specified"""
+        timezone = pytz.timezone(timezone_name) if timezone_name else self.local_tz
+        current_time = datetime.now(timezone)
+        
+        return {
+            "timezone": timezone_name or str(self.local_tz),
+            "time": current_time.strftime("%H:%M %Z"),
+            "date": current_time.strftime("%Y-%m-%d"),
+            "full_datetime": current_time.strftime("%Y-%m-%d %H:%M:%S %Z"),
+            "is_dst": bool(current_time.dst())
+        }
+
+    def convert_time(self, source_tz: str, time_str: str, target_tz: str) -> Dict[str, Any]:
+        """Convert time between timezones"""
+        try:
+            source_timezone = pytz.timezone(source_tz)
+            target_timezone = pytz.timezone(target_tz)
+            
+            # Parse time
+            hour, minute = map(int, time_str.split(":"))
+            if not (0 <= hour <= 23 and 0 <= minute <= 59):
+                raise ValueError
+        except pytz.exceptions.UnknownTimeZoneError as e:
+            raise ValueError(f"Unknown timezone: {str(e)}")
+        except:
+            raise ValueError("Invalid time format. Expected HH:MM (24-hour format)")
+        
+        # Create time in source timezone
+        now = datetime.now(source_timezone)
+        source_time = source_timezone.localize(
+            datetime(now.year, now.month, now.day, hour, minute)
+        )
+        
+        # Convert to target timezone
+        target_time = source_time.astimezone(target_timezone)
+        date_changed = source_time.date() != target_time.date()
+        
+        return {
+            "source": {
+                "timezone": str(source_timezone),
+                "time": source_time.strftime("%H:%M %Z"),
+                "date": source_time.strftime("%Y-%m-%d"),
+                "full_datetime": source_time.strftime("%Y-%m-%d %H:%M:%S %Z")
+            },
+            "target": {
+                "timezone": str(target_timezone),
+                "time": target_time.strftime("%H:%M %Z"),
+                "date": target_time.strftime("%Y-%m-%d"),
+                "full_datetime": target_time.strftime("%Y-%m-%d %H:%M:%S %Z")
+            },
+            "time_difference": f"{(target_time.utcoffset() - source_time.utcoffset()).total_seconds() / 3600:+.1f}h",
+            "date_changed": date_changed,
+            "day_relation": "next day" if date_changed and target_time.date() > source_time.date() else "previous day" if date_changed else "same day"
+        }
```

Comment:
I think it would be best if we use a datetime function or something similar to parse the format. My suggestion here is using `dateutil.parser`, allowing for automatic parsing of most functions.

One suggestion is
```suggestion
    try:
        source_timezone = pytz.timezone(source_tz)
        target_timezone = pytz.timezone(target_tz)
    except KeyError as e:
        raise ValueError(f"Unknown timezone: {e}")

    try:
        # Parse the time string
        parsed_time = datetime.strptime(time_str, "%H:%M").time()
    except ValueError:
        raise ValueError("Invalid time format. Expected HH:MM (24-hour format)")

    target_time = source_time.astimezone(target_timezone)
    time_difference = target_time - source_time

    return {
        "source": source_time.isoformat(),
        "target": target_time.isoformat(),
        "time_difference": f"{time_difference.total_seconds() / 3600:+.1f}h",
        "date_changed": source_time.date() != target_time.date()
    }
```

 Alternatively we can use 

```suggestion
def convert_time(source_tz: str, time_str: str, target_tz: str) -> dict:
    try:
        source_timezone = pytz.timezone(source_tz)
        target_timezone = pytz.timezone(target_tz)
    except KeyError as e:
        raise ValueError(f"Unknown timezone: {e}")

    try:
        # Parse the time string
        parsed_time = datetime.strptime(time_str, "%H:%M").time()
    except ValueError:
        raise ValueError("Invalid time format. Expected HH:MM (24-hour format)")

    # Create a datetime object with today's date and the parsed time
    now = datetime.now(source_timezone)
    source_time = datetime.combine(now.date(), parsed_time).replace(tzinfo=source_timezone)

    target_time = source_time.astimezone(target_timezone)
    time_difference = target_time - source_time

    return {
        "source": source_time.isoformat(),
        "target": target_time.isoformat(),
        "time_difference": f"{time_difference.total_seconds() / 3600:+.1f}h",
        "date_changed": source_time.date() != target_time.date()
    }
```
We certainly should do calculations directly on time objects to return a timedelta object that we convert.